langchain4j-jlama/src/main/java/dev/langchain4j/model/jlama/JacksonJsonCodec.java

```diff
@@ -0,0 +1,84 @@
+package dev.langchain4j.model.jlama;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+
+class JacksonJsonCodec {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .enable(SerializationFeature.INDENT_OUTPUT)
+            .registerModule(new JavaTimeModule())
+            .registerModule(new SimpleModule()
+                    .addSerializer(LocalDate.class, new LocalDateSerializer())
+                    .addDeserializer(LocalDate.class, new LocalDateDeserializer())
+                    .addSerializer(LocalTime.class, new LocalTimeSerializer())
+                    .addDeserializer(LocalTime.class, new LocalTimeDeserializer())
+                    .addSerializer(LocalDateTime.class, new LocalDateTimeSerializer())
+                    .addDeserializer(LocalDateTime.class, new LocalDateTimeDeserializer())
+            );
+
+    public static <T> T fromJson(String json, Class<T> type) {
+        try {
+            if (Map.class.isAssignableFrom(type)) {
```

Comment:
This method is used only to deserialize JSON into a `Map<String, Object>`, so it probably makes no sense to have this condition. You can make this method more specific